### PR TITLE
177768193 snap login

### PR DESCRIPTION
**Pull request by christopher-hou**
Basically the same as both the Google and Microsoft logins. Needs two additional environment variables, SNAP_CLIENT_URL and SNAP_CLIENT_SECRET. From the testing, it seems to work -- I'm still waiting to test if this actually works when attempted in the actual app but I'm still waiting on a response for Michael for that I so might as well push this now for review and wait for updates

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,10 +14,11 @@ gem 'pg', '~> 1.0'
 gem 'webpacker', '~> 4'
 
 # Login via 3rd party services.
-gem 'omniauth', '~> 2.0'
+gem 'omniauth', '~> 1.0'
```
Comment (cscoder42) [inline Gemfile:17]: Why are we rolling back the omniauth to version 1?

**Review by cscoder42 — APPROVED** [review]
Why are we rolling back omniauth? Especially since this is an entire version change, there might be security implications. Everything else looks good.

File: Gemfile (hunk 1 above)
Comment (christopher-hou) [reply Gemfile:17]: omniauth-discourse only seems to work with version 1, but if someone can make it work otherwise I'll change it back.

**Review by justinphyo — APPROVED** [review]
Looks good

**Pull request merged**